src/GitHub.App/Controllers/UIController.cs

```diff
@@ -254,7 +262,8 @@ public void Start([AllowNull] IConnection connection)
                             .PermitIf(Trigger.Clone, UIViewType.Login, () => !host.IsLoggedIn)
                             .PermitIf(Trigger.Publish, UIViewType.Publish, () => host.IsLoggedIn)
                             .PermitIf(Trigger.Publish, UIViewType.Login, () => !host.IsLoggedIn)
-                            .PermitIf(Trigger.Gist, UIViewType.Gist, () => host.IsLoggedIn)
+                            .PermitIf(Trigger.Gist, UIViewType.Gist, () => host.IsLoggedIn && host.SupportsGist)
+                            .PermitIf(Trigger.Gist, UIViewType.LogoutRequired, () => host.IsLoggedIn && !host.SupportsGist)
```

Comment:
You don't need this to be evaluated after the first time it runs, right? If you do, there's `PermitDynamic` and `PermitDynamicIf`, which allows you to chose the UIViewType to load every time it gets called (just fyi)